File: backend/Ventas.Application/Shared/SearchCommand.cs
namespace Ventas.Application.Shared
{
    public class SearchCommand
    {
        public int? Id { get; set; }
        public List<SearchFilterCommand> Filters { get; set; } = [];
        public List<string> Includes { get; set; } = [];
        public bool DisableTracking { get; set; } = true;
        public int Deleted { get; set; } = 0;
    }

    public class SearchFilterCommand
    {
        public string Field { get; set; } = string.Empty;
        public string Value { get; set; } = string.Empty;
        public List<int> Ids { get; set; } = [];
    }
}

File: frotend/src/DTOs/SearchCommand.js
export class SearchFilterCommand {
  constructor(field = "", value = "", ids = []) {
    this.field = field;
    this.value = value;
    this.ids = ids;
  }
}

export class SearchCommand {
  constructor({
    id = null,
    filters = [],
    includes = [],
    disableTracking = true,
    deleted = 0,
  } = {}) {
    this.id = id;
    this.filters = filters;
    this.includes = includes;
    this.disableTracking = disableTracking;
    this.deleted = deleted;
  }
}
